libraries/chain/controller.cpp

```diff
@@ -670,11 +676,11 @@ struct building_block {
                                 auto trx_merkle_fut =
                                    post_async_task(ioc, [&]() { return legacy_merkle(std::move(trx_receipts)); });
                                 auto action_merkle_fut =
-                                   post_async_task(ioc, [&]() { return legacy_merkle(std::move(action_receipts)); });
+                                   post_async_task(ioc, [&]() { return legacy_merkle(std::move(*action_receipts.digests_l)); });
                                 return std::make_pair(trx_merkle_fut.get(), action_merkle_fut.get());
                              },
                              [&](const checksum256_type& trx_checksum) {
-                                return std::make_pair(trx_checksum, legacy_merkle(std::move(action_receipts)));
+                                return std::make_pair(trx_checksum, legacy_merkle(std::move(*action_receipts.digests_l)));
```

Comment:
Maybe `assert()` or even add a method to `action_receipts` that asserts.

Reply:
The `std::optional` already asserts that it is not empty when dereferenced, but I can add our own too.

Reply:
Good point. Not needed then.

Reply:
OK, thanks, then I'll leave as-is.